tree/ntuple/v7/inc/ROOT/RDaos.hxx

```diff
@@ -112,17 +112,18 @@ public:
       FetchUpdateArgs() = default;
       FetchUpdateArgs(const FetchUpdateArgs &) = delete;
       FetchUpdateArgs(FetchUpdateArgs &&fua);
-      FetchUpdateArgs(DistributionKey_t d, std::span<const AttributeKey_t> as, std::span<d_iov_t> vs,
+      FetchUpdateArgs(DistributionKey_t d, std::span<const AttributeKey_t> as, std::vector<std::vector<d_iov_t>> &&vs,
                       bool is_async = false);
       FetchUpdateArgs &operator=(const FetchUpdateArgs &) = delete;
       daos_event_t *GetEventPointer();
 
       /// \brief A `daos_key_t` is a type alias of `d_iov_t`. This type stores a pointer and a length.
       /// In order for `fDistributionKey` to point to memory that we own, `fDkey` holds the distribution key.
-      /// `fAkeys` and `fIovs` are sequential containers assumed to remain valid throughout the fetch/update operation.
+      /// `fAkeys` is a sequential container assumed to remain valid throughout the fetch/update operation.
+      /// `fIovs` holds a vector of IOVs associated with requests for each attribute key in `fAkeys`.
       DistributionKey_t fDkey{};
       std::span<const AttributeKey_t> fAkeys{};
-      std::span<d_iov_t> fIovs{};
+      std::vector<std::vector<d_iov_t>> fIovs{};
```

Comment:
Maybe we can simplfiy these two as a `std::span<>` of some data structure that holds both, the attribute key + the list of IOVs associated with it for data transfer?

That might also simplify some of the code below (and maybe save some memory allocations).

Reply:
Done!